File: go/cote/LTCD/보석과_돌.go
package LTCD

import (
	"fmt"	
	"math/rand"
)

// const (
// 	TEST_CASE = 10
// 	STONES = 100
// 	JEWALS = 'z' - 'a' + 1
// )

type 보석과_돌 struct {
	TEST_CASE int 
	STONES int 
	JEWALS int
}

func (problem 보석과_돌) solve(jewals ,stones []string) int {
	ret := 0
	freq := make(map[string]int, 0)

	for _, stone := range stones {
		_, ok := freq[stone]

		if !ok {
			freq[stone] = 0
		}

		freq[stone] += 1
	}

	for _, jewal := range jewals {
		ret += freq[jewal]
	}

	return ret
}

func (problem 보석과_돌) main() {
	random := rand.New(rand.NewSource(43))

	for it := 0; it < problem.TEST_CASE; it++ {
			
		jewals := make([]string, 0)

		for c := 'a'; c < 'z'; c++ {
			if random.Float32() < 0.5 {
				jewals = append(jewals, string(c))
			}
		}

		stones := make([]string, random.Intn(problem.STONES - 1) + 1)

		for i, _ := range stones {
			stones[i] = string('a' + random.Intn(problem.JEWALS))
		}

		fmt.Println(problem.solve(jewals, stones))
	}
}

File: python/cote/LTCD/보석과_돌.py
import random
from typing import List
from collections import defaultdict

TEST_CASES = 10
JEWALS = ord("z") - ord("a") + 1
STONES = 100


def solve(jewals: List[int], stones: List[int]) -> int:
    freq = defaultdict(int)

    for stone in stones:
        freq[stone] += 1

    ret = 0
    for jewal in jewals:
        ret += freq[jewal]

    return ret


if __name__ == "__main__":

    for _ in range(TEST_CASES):

        jewals = random.sample(
            [chr(ord("a") + i) for i in range(JEWALS)], random.randrange(1, JEWALS)
        )
        stones = [random.choice([chr(ord("a") + j) for j in range(JEWALS)]) for i in range(STONES)]
        print(solve(jewals, stones))
